### ISSUE TITLE
Replace package.properties with manifest.xml

### ISSUE THREAD

**Issue by smehrbrodt** [LibreOffice/loeclipse#37]
For full ant support, we will need a working manifest.xml as it cannot be generated during runtime.
Would be nice to have an eclipse editor for that and also a way to convert package.properties to manifest.xml.

See also:
- http://bosdonnat.fr/new-export-wizard-for-ooeclipse.html
- https://wiki.openoffice.org/wiki/Documentation/DevGuide/Extensions/File_Format

**Comment by prrvchr** [comment]
Ok I read [Cedric Bosdonnat's ticket](https://bosdonnat.fr/new-export-wizard-for-ooeclipse.html). I don't think we can merge `package.properties` and `manifest.xml` file. These two files have very different purposes. One is for managing the contents of the oxt archive while the second is there for managing the UNO services, `.idl`, `.xcu` and `.xcs` files declared in the archive.

The `package.properties` file can only be managed via its Eclipse editor or a text editor, while the `manifest.xml` file should be managed automatically. It must be created during the initial creation of an extension (the xml file with its root element) and must be updated automatically during:
- The creation, modification and deletion of the Java or Python UNO service (this is what the new passive registration does).
- The compilation of`.idl` files in the background (this remains to be implemented).
- The creation and deletion of `.xcu` and/or `.xcs` xml files (this remains to be implemented).
- The declaration of Dialog or Basic Libraries (this remains to be implemented).

Perhaps there are other files to consider when updating the `manifest.xml` file that I may have forgotten?

Please confirm or deny.

**Comment by smehrbrodt** [comment]
Yeah makes sense to keep the `package.properties` as it is, and the `manifest.xml` as a generated file.

Probably the `package.properties` to `manifest.xml` converter should be an own package so that it can be used from ant and other build tools?

**Comment by prrvchr** [comment]
I think we should view the contents of an extension's .oxt file as a combination of:
- The `contents` property of the `package.properties` file.
- The value of the `full-path` attribute of the `file-entry` elements of the `META-INF/manifest.xml` file.
- If the `package.components` is referenced in `META-INF/manifest.xml`, the value of its `uri` attribute of its `component` elements.

If we combine all the files referenced in these three different files, we should be able to obtain the exact contents of an extension's .oxt file.

This requires parsing both manifest.xml and package.components xml files to get the list of files to include in the oxt file. And we would need to do the same with **Ant**...
